### DISCUSSION TITLE
Update UI every X seconds using MVUX

### DISCUSSION THREAD

**Discussion by LuisAlfredo92** [unoplatform/uno#13357] [Q&A]
I have a class with some information that will be shown in cards, something like this:
```
public record MyClass
{
	public string Name { get; init; }
	public int Id { get; init; }
	public CodeGenerator Code { get; init; }
	public int Duration { get; init; }
	
	public MyClass(string name, int id, int? duration = null)
	{
		Name = name;
		Id = id;
		Duration = duration ?? 60;
		Code = new CodeGenerator(Duration);
	}
}
```
All this info is taken from a database

That `Code` object generates codes that are updated every `Duration` seconds, 60 seconds by default, and I have to show the code in the card, however, it calculates them only calling a `Generate()` method inside of it and returns a string

So, **how can I update that code in UI using MVUX?**

- I tried to bind the function in XAML, but it doesn't display anything
- I tried to add a new property and update it with a new task like this:
```
.
.
.
// In properties
public string GeneratedCode { get; private set; }
.
.
.
// In constructor
ExecuteEveryXSeconds(Duration, () => {GeneratedCode = Code.Generate();});
```
but it didn't work because UI uses inmutable objects, so the code is updated but not shown in UI
- Tried to use `RefreshContainer` and `RefreshView` but [docs don't say anything about them](https://platform.uno/docs/articles/implemented/windows-ui-xaml-controls-refreshcontainer.html)
- I saw something about a `Run` XAML tag [in the gallery](https://gallery.platform.uno/), but I didn't find more information

Could you help me please?

**Comment by limeniye** [comment]
Hello @LuisAlfredo92!
I saw something by @nickrandolph 
https://nicksnettravels.builttoroam.com/feeds-in-mvux/

**Reply by LuisAlfredo92** [reply]
Ooohh, thanks

I use one feed to show the cards but I didn't know they could be used to update values like that, I'll try to implement them, thanks!

**Reply by limeniye** [reply]
Also, I`am not sure that this ```ExecuteEveryXSeconds(Duration, () => {GeneratedCode = Code.Generate();});``` will update your view. Because for MVUX you need to change the full object and not the other way around.
https://github.com/unoplatform/uno.extensions/discussions/1163#discussioncomment-5053659
Simple example:
```C#
public MyClass MyClassObservableProperty 
{ 
    get => ...;
    set => ... ; 
}
```
```C#
private void UpdateMyClassObservableProperty()
{
    var generatedCodeSomeTestValue = ... ;
    MyClassObservableProperty = MyClassObservableProperty with { GeneratedCode = generatedCodeSomeTestValue };
}
```
```C#
ExecuteEveryXSeconds(5, () => UpdateMyClassObservableProperty);
```

If you can add some simple solution, I would be happy to check it.

**Reply by LuisAlfredo92** [reply]
Thanks for the replies, I've  been trying some things based on what you write but it doesn't work yet

I'll be trying to make it work, thanks for the help
